bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/launch/LaunchUtils.java

```diff
@@ -145,9 +147,33 @@ public static void initConfigurableVariables(Path configFilePath, Map<Module, Va
         }
     }
 
-    public static Path getConfigPath() {
-        Map<String, String> envVariables = System.getenv();
-        return Paths.get(envVariables.getOrDefault(CONFIG_ENV_VARIABLE, Paths.get(RuntimeUtils.USER_DIR,
-                                                                                  CONFIG_FILE_NAME).toString()));
+    private static List<ConfigProvider> getConfigProviders(Path[] configFilePaths,
+                                                           Map<Module, VariableKey[]> configurationData) {
+        List<ConfigProvider> supportedConfigProviders = new LinkedList<>();
+        if (configFilePaths.length == 1 && configFilePaths[0].equals(DEFAULT_CONFIG_PATH)) {
+            String configData = System.getenv().get(CONFIG_DATA);
```

Comment:
This lookup has happen on the main side, otherwise testarina values can also be overridden using this.